src/phoenix/server/api/dataloaders/document_retrieval_metrics.py

```diff
@@ -0,0 +1,91 @@
+from collections import defaultdict
+from itertools import groupby
+from typing import (
+    AsyncContextManager,
+    Callable,
+    DefaultDict,
+    Dict,
+    List,
+    Optional,
+    Set,
+    Tuple,
+)
+
+import numpy as np
+from sqlalchemy import select
+from sqlalchemy.ext.asyncio import AsyncSession
+from strawberry.dataloader import DataLoader
+from typing_extensions import TypeAlias
+
+from phoenix.db import models
+from phoenix.metrics.retrieval_metrics import RetrievalMetrics
+from phoenix.server.api.types.DocumentRetrievalMetrics import DocumentRetrievalMetrics
+
+RowId: TypeAlias = int
+NumDocs: TypeAlias = int
+EvalName: TypeAlias = Optional[str]
+Key: TypeAlias = Tuple[RowId, NumDocs, EvalName]
+
+
+class DocumentRetrievalMetricsDataLoader(DataLoader[Key, List[DocumentRetrievalMetrics]]):
+    def __init__(self, db: Callable[[], AsyncContextManager[AsyncSession]]) -> None:
+        super().__init__(load_fn=self._load_fn)
+        self._db = db
+
+    async def _load_fn(self, keys: List[Key]) -> List[List[DocumentRetrievalMetrics]]:
+        results: Dict[Key, List[DocumentRetrievalMetrics]] = {key: [] for key in keys}
+        requests: DefaultDict[RowId, DefaultDict[NumDocs, Set[EvalName]]] = defaultdict(
+            lambda: defaultdict(set)
+        )
+        for row_id, num_docs, eval_name in results.keys():
+            requests[row_id][num_docs].add(eval_name)
+        mda = models.DocumentAnnotation
```

Comment:
I think the alias is appropriate in this case, because there is only one table being referenced, so there should be no confusion, and repeating the same long-form over and over (for 11 times) just creates unnecessary visual clutter.